frontend/components/Header/Nav/MainMenu/Column.js

```diff
@@ -0,0 +1,358 @@
+// @flow
+import { Component } from 'react';
+import { headline, egyptian } from '@guardian/pasteup/fonts';
+
+import { css, cx } from 'react-emotion';
+import { pillars } from '@guardian/pasteup/palette';
+import { desktop, tablet, leftCol } from '@guardian/pasteup/breakpoints';
+
+const hideDesktop = css`
+    ${desktop} {
+        display: none;
+    }
+`;
+const pillarColor = pillar =>
+    css`
+        color: ${pillars[pillar]};
+    `;
+
+const showColumnLinksStyle = css`
+    :before {
+        margin-top: 8px;
+        transform: rotate(-135deg);
+    }
+`;
+const hideAfter = css`
+    :after {
+        display: none;
+    }
+`;
+
+const colapseColumnButton = css`
+    background-color: transparent;
+    border: 0;
+    box-sizing: border-box;
+    cursor: pointer;
+    display: block;
+    font-family: ${headline};
+    font-size: 24px;
+    font-weight: 700;
+    line-height: 1;
+    outline: none;
+    padding: 6px 34px 18px 50px;
+    position: relative;
+    text-align: left;
+    width: 100%;
+    > * {
+        pointer-events: none;
+    }
+    text-transform: capitalize;
+    :before {
+        margin-top: 4px;
+        color: #5d5f5f;
+        left: 25px;
+        position: absolute;
+        border: 2px solid currentColor;
+        border-top: 0;
+        border-left: 0;
+        content: '';
+        display: inline-block;
+        height: 10px;
+
+        transform: rotate(45deg);
+        width: 10px;
+    }
+
+    :after {
+        background-color: #abc2c9;
+        bottom: 0;
+        content: '';
+        display: block;
+        height: 1;
+        left: 50;
+        position: absolute;
+        width: 100%;
+    }
+`;
+
+const ColapseColumnButton = ({
+    column,
+    showColumnLinks,
+    toggleColumnLinks,
+    ariaControls,
+    isLastIndex,
+}: {
+    column: LinkType,
+    showColumnLinks: boolean,
+    toggleColumnLinks: () => void,
+    ariaControls: string,
+    isLastIndex: boolean,
+}) => (
```

Comment:
According to the [flow docs](https://flow.org/en/docs/react/types/#toc-react-element) the way to type stateless components is: 
```js
const element: React.Element<'div'> = <div />;
```
However as mentioned in https://github.com/facebook/flow/issues/4130

> `import React from 'react'` doesn't exposes the React types anymore. So if you want to use `React.Element<*>`, you need to use `import * as React from 'react'`.

OR this should work:
```js
const element: React$Element<*> = <div />;
```

Reply:
To me this is a benefit of typescript.

Reply:
I think how we type React components like this probably merits wider discussion.